datatables/datatables.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import absolute_import
+
+import math
+
+from sqlalchemy import Text, func, or_
+from sqlalchemy.dialects import mysql, postgresql, sqlite
+
+from datatables.clean_regex import clean_regex
+from datatables.search_methods import SEARCH_METHODS
+
+
+class DataTables:
+    """Class defining a DataTables object.
+
+    :param request: request containing the GET values, specified by the
+        datatable for filtering, sorting and paging
+    :type request: pyramid.request
+    :param query: the query wanted to be seen in the the table
+    :type query: sqlalchemy.orm.query.Query
+    :param columns: columns specification for the datatables
+    :type columns: list
+
+    :returns: a DataTables object
+    """
+
+    def __init__(self, request, query, columns, allow_regex_searches=False):
+        """Initialize object and run the query."""
+        self.params = dict(request)
+        if 'sEcho' in self.params:
+            raise ValueError(
+                'Legacy datatables not supported, upgrade to >=1.10')
+        self.query = query
+        self.columns = columns
+        self.results = None
+        self.allow_regex_searches = allow_regex_searches
+
+        # total in the table after filtering
+        self.cardinality_filtered = 0
+
+        # total in the table unfiltered
+        self.cardinality = 0
+
+        self.yadcf_params = []
+        self.filter_expressions = []
+        self.error = None
+        try:
+            self.run()
+        except Exception as exc:
+            self.error = str(exc)
+
+    def output_result(self):
+        """Output results in the format needed by DataTables."""
+        output = {}
+        output['draw'] = str(int(self.params['draw']))
```

Comment:
Might as wel include the fix discussed in #98 

```suggestion
        output['draw'] = str(int(self.params.get('draw',1))
```